src/timesync/timesyncd-manager.c

```diff
@@ -1030,7 +1030,11 @@ static int manager_network_event_handler(sd_event_source *s, int fd, uint32_t re
 
         sd_network_monitor_flush(m->network_monitor);
 
-        changed = !!manager_network_read_link_servers(m);
+        r = manager_network_read_link_servers(m);
+        if (r < 0)
+                return r;
```

Comment:
hmm, i don't think we should propagate the error here, note that sd-event handler functions which return a failure will cause the event source they are registered with to be disabled. I don't think that's a good choice here. Hence, we should log about errors here, and exit the function, but not propagate the error, but rather return 0, so that the event source stays enabled, after all the error here is very much recoverable I guess.

Reply:
OK. I've dropped the change, and add a comment.